### PR TITLE
CLDR-15156 Avoid getting stuck, Waiting for the SurveyTool to start up

### PR DESCRIPTION
**Pull request by btangmu**
-On the front end, prohibit # = U+0023 NUMBER SIGN in id

CLDR-15156

- [ ] This PR completes the ticket.

### CODE REVIEW

File: tools/cldr-apps/js/test/mockup.html [hunk 1]
```diff
@@ -188,9 +188,7 @@
           <li>Survey Tool 40 Data Submission</li>
           <li><a class="gear" href="#gear">⚙</a></li>
           <li>
-            <a
-              href="https://cldr.unicode.org/translation/"
-              target="_blank"
+            <a href="https://cldr.unicode.org/translation/" target="_blank"
```
Comment (btangmu) [inline tools/cldr-apps/js/test/mockup.html:191]: this is just auto-formatting

File: tools/cldr-apps/js/src/esm/cldrStatus.js [hunk 2]
```diff
@@ -121,7 +120,21 @@ function getCurrentId() {
...
+  currentId = idIsAllowed(id) ? id : "";
+}
+
+function idIsAllowed(id) {
+  if (!id) {
+    return false;
+  }
+  /*
+   * The hash character # (U+0023 NUMBER SIGN) is not allowed in the id; probably
+   * other URL reserved characters should not be allowed either
```
Comment (srl295) [inline tools/cldr-apps/js/src/esm/cldrStatus.js:132]: could be `[0-9a-fA-F]` (hex) only

**Review by srl295 — APPROVED** [review]
LGTM. Not a ST bug per se, but adds robustness!

**Comment by btangmu** [comment]
@srl295 "could be [0-9a-fA-F] (hex) only" -- maybe so; forum id is always numeric, right? I'm not sure if that accounts for every kind of id though

**Comment by srl295** [comment]
Oh I thought this was strid

**Comment by srl295** [comment]
Oh I thought this was strid

**Comment by btangmu** [comment]
> Oh I thought this was strid

various "specials" also use an id in the url, and some of them (at least forum) use cldrStatus.get/setCurrentId

**Pull request merged**